GVRf/Framework/src/org/gearvrf/GVRScene.java

```diff
@@ -152,6 +153,66 @@ public void setFrustumCulling(boolean flag) {
     public void setOcclusionQuery(boolean flag) {
         NativeScene.setOcclusionQuery(getNative(), flag);
     }
+
+    private GVRConsole mStatsConsole = null;
+    private boolean mStatsEnabled = false;
+    private boolean pendingStats = false;
+
+    /**
+     * Returns whether displaying of stats is enabled for this scene.
+     * 
+     * @return whether displaying of stats is enabled for this scene.
+     */
+    public boolean getStatsEnabled() {
+        return mStatsEnabled;
+    }
+
+    /**
+     * Set whether to enable display of stats for this scene.
+     *
+     * @param enabled
+     *     Flag to indicate whether to enable display of stats.
+     */
+    public void setStatsEnabled(boolean enabled) {
```

Comment:
ok, changed the way this works to match what you suggested.  will have to investigate the performance problem later today.